portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -2041,57 +2045,55 @@ public void init(ServletConfig config) throws ServletException
 		
 		sakaiTutorialEnabled = ServerConfigurationService.getBoolean("portal.use.tutorial", true);
 
-		// Load the related links.
+		// Load the related links if they are in the properties file.
 		List<String> linkUrls = Arrays.asList(emptyNotNull(ServerConfigurationService.getStrings("related.link.url")));
 		List<String> linkTitles = Arrays.asList(emptyNotNull(ServerConfigurationService.getStrings("related.link.title")));
 		List<String> linkNames = Arrays.asList(emptyNotNull(ServerConfigurationService.getStrings("related.link.name")));
 		List<String> linkIcons = Arrays.asList(emptyNotNull(ServerConfigurationService.getStrings("related.link.icon")));
-		
-		List<Map> relatedLinks = new ArrayList<Map>(linkUrls.size());
-		for (int i = 0; i < linkUrls.size(); i++)
-		{
-			String url = linkUrls.get(i);
-			String title = linkTitles.get(i);
-			String name = linkNames.get(i);
-			String icon = linkIcons.get(i);
-			if (url != null)
-			{
-				Map<String,String> linkDetails = new HashMap<String,String>();
-				linkDetails.put("url", url);
-				if (name != null)
-				{
-					linkDetails.put("name", name);
-					if (title != null)
-					{
-						linkDetails.put("title", title);
-					}
-					else
-					{
-						linkDetails.put("title", name);
-					}
-				}
-				else
-				{
-					if (title != null)
-					{
-						linkDetails.put("name", title);
-						linkDetails.put("title", title);
+
+		if (!linkUrls.isEmpty()) {
+			List<Map> relatedLinks = new ArrayList<Map>(linkUrls.size());
+			for (int i = 0; i < linkUrls.size(); i++) {
+				String url = linkUrls.get(i);
+				String title = linkTitles.get(i);
+				String name = linkNames.get(i);
+				String icon = linkIcons.get(i);
+
+				if (url != null) {
+					Map<String, String> linkDetails = new HashMap<String, String>();
+					linkDetails.put("url", url);
+					if (name != null) {
+						linkDetails.put("name", name);
+						if (title != null) {
+							linkDetails.put("title", title);
+						} else {
+							linkDetails.put("title", name);
+						}
+					} else {
+						if (title != null) {
+							linkDetails.put("name", title);
+							linkDetails.put("title", title);
+						} else {
+							linkDetails.put("name", url);
+							linkDetails.put("title", url);
+						}
 					}
-					else
-					{
-						linkDetails.put("name", url);
-						linkDetails.put("title", url);
+					if (icon != null) {
+						// first two characters in the properties file indicate whether it is an icon or image.
+						if (icon.substring(0,2).equalsIgnoreCase("im")) {
+							linkDetails.put("iconType", "image");
+							linkDetails.put("imageURI", icon.substring(3));
```

Comment:
This will blow up with a IndexOutOfBoundsException if the value is "im" or if the value is anything less that 4 characters long.